src/cmds/qstat.c

```diff
@@ -134,6 +137,87 @@ static struct attrl alt_attribs[] = {
 
 static struct attrl *display_attribs = &basic_attribs[0];
 
+int cmp_jobs(const void *j1, const void *j2)
+{
+	char *job1, *job2;
+	char server1[MAXSERVERNAME], server2[MAXSERVERNAME];
+	char job_id[PBS_MAXCLTJOBID];
+	char *seq_num1 = NULL;
+	char *seq_num2 = NULL;
+	char *job_id_out = NULL;
+	char *pserver_one = NULL;
+	char *pserver_two = NULL;
+	char *current_server = NULL;
+	int ret = 0;
+	int ret_val = 0;
+	char *eptr = NULL;
+	long jid1,jid2;
+	job1 = *(char **)j1;
+	job2 = *(char **)j2;
+	if (pbs_isjobid(job1)== 0) {
+		ret_val = 1;
+		goto return_here;
+	}
+	else if (pbs_isjobid(job2) == 0) {
+		ret_val = 1;
+		goto return_here;
+	}
+	else {
+		parse_jobid(job1, &seq_num1, &pserver_one, &current_server);
```

Comment:
Hi, why are we setting current_server if it is not going to get used. We should pass null in place of current_server and
parse_jobid should handle it accordingly.